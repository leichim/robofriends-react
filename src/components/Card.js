import React from 'react';
import { shallow, mount, render } from 'enzyme';

// A JS function implies it being the render function
const Card = ({id, name, email}) => {
    return (
        <div className="tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5">
            <img src={'https://robohash.org/' + id + '?height=200width=200'} alt="Robots" />
            <div>
                <h2>{name}</h2>
                <p>{email}</p>
            </div>
        </div>
    )
}

export default Card;