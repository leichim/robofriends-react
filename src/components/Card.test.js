import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Card from './Card';

/** 
 * Shallow rendering is just rendering the card component (not any children components inside the component)
 * A form of unit testing for components
 * 
 * Mount is used for components that may interact with the dom, but you need a browser or jsdom because it does a full render (but it's rarely used)
 * 
 * Render is used to render react components, but unlike render it is rendered to a static html using cheer.io
 */
it('render card component', () => {
    // console.log(shallow(<Card />));
    // expect( shallow(<Card />).length ).toEqual(1);
    expect( shallow(<Card />) ).toMatchSnapshot(); // Detects any changes on the card, saves a snapshot in __snaphots__
});