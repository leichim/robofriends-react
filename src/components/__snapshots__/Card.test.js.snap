// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render card component 1`] = `ShallowWrapper {}`;
